### PR TITLE
Updated release script documentation and command names

### PR DESCRIPTION
**Pull request by bvaughn**
Updated our team's release script documentation and command names to match what our [public release channel process](https://reactjs.org/docs/release-channels.html) says.

- [x] Update release script README to match new process
- [x] Remove all references to "canary" since it's overloaded at this point.
- [x] Run through all of the command steps after making the changes (with `--dry` on for publishing) to make sure everything still works.
- [x] Update example build ids and release version numbers to ones that actually work (given more recent changes to build pipeline).

### CODE REVIEW

File: scripts/release/README.md [hunk 1]
```diff
@@ -19,66 +19,71 @@ The high level process of creating releases is [documented below](#process). Ind
...
+scripts/release/prepare-stable.js --version=0.0.0-241c4467e
 ```
 
 This script will prompt you to select stable version numbers for each of the packages. It will update the package JSON versions (and dependencies) based on the numbers you select.
 
 Once this step is complete, you're ready to publish the release:
 
 ```sh
-scripts/release/publish.js --tags next latest
+scripts/release/publish.js --tags latest
```
Comment (bvaughn) [inline scripts/release/README.md]: Note this is arguably a significant change. If "next" releases are identified as always being 0.0.0-<hash>, and "latest" releases are always semver numbers, it doesn't make sense for us to update `next` to match `latest` on a stable release. I think that's fine though.

File: scripts/release/publish-commands/validate-tags.js [hunk 2]
```diff
@@ -21,14 +21,14 @@ const run = async ({cwd, packages, tags}) => {
...
+        theme`{error Next release} {version ${version}} {error cannot be tagged as} {tag latest}`
       );
       process.exit(1);
     }
   } else {
-    if (tags.includes('canary')) {
+    if (tags.includes('next')) {
       console.log(
-        theme`{error Stable release} {version ${version}} {error cannot be tagged as} {tag canary}`
+        theme`{error Stable release} {version ${version}} {error cannot be tagged as} {tag next}`
```
Comment (bvaughn) [inline scripts/release/publish-commands/validate-tags.js]: This goes with my other comment about preventing us from tagging a semver range as `next`.

File: scripts/release/README.md [hunk 3]
```diff
@@ -4,12 +4,12 @@ The release process consists of several phases, each one represented by one of t
 
 A typical release goes like this:
 1. When a commit is pushed to the React repo, [Circle CI](https://circleci.com/gh/facebook/react/) will build all release bundles and run unit tests against both the source code and the built bundles.
-2. Next the release is [**published as a canary**](#publishing-a-canary) using the [`prepare-canary`](#prepare-canary) and [`publish`](#publish) scripts. (Currently this process is manual but might be automated in the future using [GitHub "actions"](https://github.com/features/actions).)
-3. Finally, a canary releases can be [**promoted to stable**](#publishing-a-stable-release)<sup>1</sup> using the [`prepare-stable`](#prepare-stable) and [`publish`](#publish) scripts. (This process is always manual.)
+2. Next the release is [**published to the `next` channel**](#publishing-next) using the [`prepare-next`](#prepare-next) and [`publish`](#publish) scripts. (Currently this process is manual but might be automated in the future using [GitHub "actions"](https://github.com/features/actions).)
```
Comment (threepointone) [inline scripts/release/README.md]:
```suggestion
2. The release is then [**published to the `next` channel**](#publishing-next) using the [`prepare-next`](#prepare-next) and [`publish`](#publish) scripts. (Currently this process is manual but might be automated in the future using [GitHub "actions"](https://github.com/features/actions).)
```

File: scripts/release/README.md [hunk 4]
```diff
@@ -96,53 +101,53 @@ Next cherry pick any changes from master that you want to include in the release
...
 ```
 
 ## `prepare-stable`
-Checks out a canary release from NPM and prepares it to be published as a stable release.
+Checks out a "next" release from NPM and prepares it to be published as a stable release.
 
 This script prompts for new (stable) release versions for each public package and updates the package contents (both `package.json` and inline version numbers) to match. It also updates inter-package dependencies to account for the new versions.
 
-Canary release have already been tested but it is still a good idea to **manually test and verify a release** before publishing to ensure that e.g. version numbers are correct. Upon completion, this script prints manual testing instructions.
+Next release have already been tested but it is still a good idea to **manually test and verify a release** before publishing to ensure that e.g. version numbers are correct. Upon completion, this script prints manual testing instructions.
```
Comment (threepointone) [inline scripts/release/README.md]:
```suggestion
This release has already been tested, but it is still a good idea to **manually test and verify a release** before publishing to ensure that e.g. version numbers are correct. Upon completion, this script prints manual testing instructions.
```

**Pull request merged**